src/dataset.jl

```diff
@@ -189,10 +194,12 @@ will be deleted and replaced with `samples`.
 function store!(dataset::Dataset, uuid::UUID, name::Symbol,
                 samples::Samples; overwrite::Bool=true)
     recording, signal = dataset.recordings[uuid], samples.signal
-    if haskey(recording.signals, name)
-        @assert overwrite "$name already exists in $uuid and `overwrite` is `false`"
+    if haskey(recording.signals, name) && !overwrite
+        throw(ArgumentError("$name already exists in $uuid and `overwrite` is `false`"))
+    end
+    if !is_lower_snake_case_alphanumeric(string(name))
+        throw(ArgumentError("$name is not lower snake case and alphanumeric"))
     end
-    @assert is_valid(signal) && is_lower_snake_case_alphanumeric(string(name))
```

Comment:
I removed the `is_valid(signal)` check because it should already be valid by construction, since it's already part of a `Samples` object. If that should be kept, it probably makes sense to have that be an assertion.

Reply:
IMO we should keep this and have it be an `ArgumentError` like the others; I actually had it here on purpose even though it shouldn't be an assertion (for the same reasons as all the others).

Onda attempts to follow the "be liberal in what you read in, but be strict about what you write back out" doctrine. In other words, we want people to be able to load in ill-formatted datasets even if e.g. the signal names are wonky, so we wouldn't want this check in the `Signal` constructor; it would just make it harder to fix invalid data. This check makes sense here, however, because we want to encourage people to fix invalid data before writing it out (especially if it might overwrite valid data). 

Technically, it's probably a good idea to both keep this check here (in order to error earlier) and have a duplicate (but probably disable-able) check against all signals in `save_recordings_file`. That definitely doesn't need to happen as part of this PR, though.